apps/api/src/secret/service/secret.service.ts

```diff
@@ -490,6 +490,45 @@ export class SecretService {
 
     return response
   }
+  async getRevisionsOfSecret(
+    user: User,
+    secretId: Secret['id'],
+    environmentId: Environment['id'],
+    page: number,
+    limit: number,
+    sort: string,
+    order: string,
+    search: string
+  ) {
+    //check access to secret
+    await this.authorityCheckerService.checkAuthorityOverSecret({
+      userId: user.id,
+      entity: { id: secretId },
+      authority: Authority.READ_SECRET,
+      prisma: this.prisma
+    })
+
+    await this.authorityCheckerService.checkAuthorityOverEnvironment({
+      userId: user.id,
+      entity: { id: environmentId },
+      authority: Authority.READ_ENVIRONMENT,
+      prisma: this.prisma
+    })
+
+    // get the revisions
+    const revisions = await this.prisma.secretVersion.findMany({
+      where: {
+        secretId: secretId,
+        environmentId: environmentId
+      },
+      skip: page * limit,
+      take: limit,
+      orderBy: {
+        [sort]: order
```

Comment:
Actually, sort this based on the version number. Get rid of the `order` and `search` fields. We would only accept the sort order.

After that, we can mark the PR as done! Although we cant merge it yet.

Reply:
Removed em !

Reply:
sort has default version,and ig it can take other field values
That okay?

Reply:
ahh wait Im getting confused by these comments on git.I have made a change rn,is it wrong ??